libs/sdk-backend-tiger/tests/integrated/backend.ts

```diff
@@ -66,5 +72,57 @@ export function testBackend(): IAnalyticalBackend {
 
 export function testWorkspace(): string {
     // UI SDK Reference workspace ID
-    return "4dc4e033e611421791adea58d34d958c";
+    const workspace: string = process.env.WORKSPACE_ID ?? getRecordingsWorkspaceId();
+    return workspace;
+}
+
+function getRecordingsWorkspaceId() {
+    return "2b7da2afb0d34f4397481c4d2a2d50b0";
+}
+
+export function sanitizeKeyWithNewValue(result: any, key: string, newValue: string) {
+    const fixtureContent = JSON.stringify(result);
+    const newResult = JSON.parse(fixtureContent, (k, v) => (k === key ? newValue : v));
+    return newResult;
+}
+
+function sanitizeWorkspaceId(result: any) {
+    if (process.env.WORKSPACE_ID) {
+        const workspace: string = process.env.WORKSPACE_ID;
+        const dataString = JSON.stringify(result, null, 2).replace(
+            new RegExp(workspace, "g"),
+            getRecordingsWorkspaceId(),
+        );
+        return JSON.parse(dataString);
+    }
+    return result;
+}
+
+export function sanitizeWorkspace(result: any, key: string, newValue: string) {
+    const newResult = sanitizeKeyWithNewValue(result, key, newValue);
+    return sanitizeWorkspaceId(newResult);
+}
+
+export function sortToOrder(json: any) {
```

Comment:
I'm thinking if we can do it more readable and type safe.
And also avoid leaking of the `TigerWorkspaceCatalog` internals into the snapshots.
What about something like (not tested it, just for an illustration):

```
import sortBy from "lodash/sortBy";

function exhaustiveCheck(_param: never): never {
    throw new Error("unknown catalog type");
}

function getCatalogItemId(item: CatalogItem) {
    if (isCatalogAttribute(item)) {
        return item.attribute.id;
    } else if (isCatalogFact(item)) {
        return item.fact.id;
    } else if (isCatalogMeasure(item)) {
        return item.measure.id;
    } else if (isCatalogDateDataset(item)) {
        return item.dataSet.id;
    } else {
        // Does the following line throw a TypeScript error?
        // You may have added a new catalog item type.
        // Cover the new item type in this function.
        return exhaustiveCheck(item);
    }
}

function sanitizeCatalogDateDataSet(dateDataset: ICatalogDateDataset): ICatalogDateDataset {
   return {
        ...dateDataset,
        dateAttributes: sortBy(dateDataset.dateAttributes, "id"),
    };
}

export function sanitizeCatalogResult(catalog: IWorkspaceCatalog) {
    const sanitizedItems = catalog
        .allItems()
        .map(item => isCatalogDateDataset(item)
            ? sanitizeCatalogDateDataSet(item)
            : item
        );

    const sortedItems = sortBy(sanitizedItems, getCatalogItemId);
    return sanitizeWorkspaceId(sortedItems);
}
```

Reply:
can't make the catalog result in order after apply above code, I'll continue it in follow-up ticket